svgnative/src/SVGRenderer.cpp

```diff
@@ -0,0 +1,66 @@
+#include "svgnative/SVGRenderer.h"
+#include "Interval.h"
+#include <tuple>
+#include <cmath>
+#include <stdexcept>
+
+namespace SVGNative
+{
+    Rect::Rect(float aX, float aY, float aWidth, float aHeight)
+    {
+        if (aWidth < 0 || aHeight < 0)
+        {
+            throw std::invalid_argument("Height or Width of a rectangle can't be negative!");
+        }
+        x = aX;
+        y = aY;
+        width = aWidth;
+        height = aHeight;
```

Comment:
We should use mX, mY, mWidth and mHeight for the members.

Reply:
Yes, we need to use like mX ,mY but due to public nature of these variables, these are used many where in code so if we do this change, we need to change in multiple files and functions.For now, i kept as it is .